test/distrib/bazel/test_single_bazel_version.sh

```diff
@@ -51,12 +51,15 @@ FAILED_TESTS=""
 export OVERRIDE_BAZEL_VERSION="$VERSION"
 # when running under bazel docker image, the workspace is read only.
 export OVERRIDE_BAZEL_WRAPPER_DOWNLOAD_DIR=/tmp
-bazel build -- //... "${EXCLUDED_TARGETS[@]}" || FAILED_TESTS="${FAILED_TESTS}Build "
+# clean the caches and downloaded archives to make sure we build from scratch
+tools/bazel clean --expunge
+tools/bazel build -- //... "${EXCLUDED_TARGETS[@]}" || FAILED_TESTS="${FAILED_TESTS}Build "
+# tools/bazel build -- //:grpc "${EXCLUDED_TARGETS[@]}" || FAILED_TESTS="${FAILED_TESTS}Build "
 
 for TEST_DIRECTORY in "${TEST_DIRECTORIES[@]}"; do
   pushd "test/distrib/bazel/$TEST_DIRECTORY/"
 
-  bazel test --test_output=all //:all || FAILED_TESTS="${FAILED_TESTS}${TEST_DIRECTORY} Distribtest"
+  tools/bazel test --cache_test_results=no --test_output=all //:all || FAILED_TESTS="${FAILED_TESTS}${TEST_DIRECTORY} Distribtest"
```

Comment:
pwd is changed so tools/bazel might not work here.

Reply:
There is a copied `tools/bazel` at https://github.com/grpc/grpc/blob/7942f30f6301f5baf93ea242d13433349640fc57/test/distrib/bazel/cpp/tools/bazel